test/e2e/monitoring_stack_controller_test.go

```diff
@@ -355,6 +358,32 @@ func assertAlertmanagerReceivesAlerts(t *testing.T) {
 	}
 }
 
+func cleanupGrafanaDS(t *testing.T) {
+	ms := newMonitoringStack(t, "test-cleanup-grafana-ds")
+	datasourceName := fmt.Sprintf("ms-%s-%s", ms.Namespace, ms.Name)
+
+	err := f.K8sClient.Create(context.Background(), ms)
+	assert.NilError(t, err, "failed to create a monitoring stack")
+
+	grafanaDS := &grafanav1alpha1.GrafanaDataSource{}
+
+	f.GetResourceWithRetry(t, datasourceName, "monitoring-stack-operator", grafanaDS)
+
+	assert.NilError(t, err, "grafana data source is not created")
+
+	f.K8sClient.Delete(context.Background(), ms)
+
+	if err = wait.Poll(5*time.Second, 20*time.Second, func() (done bool, err error) {
```

Comment:
can't we use `AssertResourceNeverExists` here?

Reply:
Done